datastructure.c

```diff
@@ -32,7 +32,8 @@ int findOverTimeID(int overTimetimestamp)
 	int timeidx = -1, i;
 	// Check struct size
 	memory_check(OVERTIME);
-	validate_access("overTime", counters.overTime-1, true, __LINE__, __FUNCTION__, __FILE__);
+	if(counters.overTime > 0)
+		validate_access("overTime", counters.overTime-1, true, __LINE__, __FUNCTION__, __FILE__);
 	for(i=0; i < counters.overTime; i++)
```

Comment:
This loop is skipped when `counters.overTime == 0`. Hence, there is also no need to check `validate_access()` before it